pkgs/top-level/all-packages.nix

```diff
@@ -21332,7 +21332,13 @@ with pkgs;
   };
 
   influxdb = callPackage ../servers/nosql/influxdb { };
-  influxdb2 = callPackage ../servers/nosql/influxdb2 { };
+  influxdb2-server = callPackage ../servers/nosql/influxdb2 {
+    buildGoModule = buildGo117Module;
+  };
+  influxdb2-cli = callPackage ../servers/nosql/influxdb2/cli.nix {
+    buildGoModule = buildGo117Module;
+  };
+  influxdb2 = callPackage ../servers/nosql/influxdb2/combined.nix { };
```

Comment:
Two small points:

* We should also use `influxdb2-server` in `influxdb2` NixOS module;
* Maybe add an explicit comment here stating that this is for backwards compatibility? I expect us to drop it in some release.

Reply:
Both done. I also added the package split as a "notable change" in the 22.05 release notes.